README_MACOS.md

```diff
@@ -35,87 +35,117 @@ Inside that application's folder (`SuperCollider.app/Contents/MacOS/`) are the t
 
 Get a copy of the source code:
 
-    git clone --recursive https://github.com/SuperCollider/SuperCollider.git
+    $> git clone --recursive https://github.com/SuperCollider/SuperCollider.git
+
+With `--recursive` all submodules are initialized and downloaded right away.
+
+**Note** The supercollider source embedds multiple dependencies as "submodule". Submodules are downloaded from a separate URL and need to be kept in sync with the current supercollider source checkout. Unfortunatelly this occasionally requires manual adjustments, as different SC revisions may require different sets and revisions of submodules (check `git submodule` documentation for a complete list of options).
+
+- initialize newly added submodules (this does not happen automatically):
+
+      $> git submodule init
+
+- get the currently required revision of a submodule that has already been initialized:
+
+      $> git submodule update
+
+- refresh the remote URL of initialized submodules (only rarely needed):
+
+      $> git submodule sync
+
+A command that combines initializing missing submodules and checking out the required revision:
+
+      $> git submodule update --init
 
-`--recursive` specifies that it should also clone the git submodules.
 
 Build instructions
 ------------------
 
-    cd SuperCollider
-    mkdir -p build
-    cd build
-    cmake -G Xcode -DCMAKE_PREFIX_PATH=`brew --prefix qt5`  ..
-    cmake --build . --target install --config RelWithDebInfo
-
-If successful this will build the application into `build/Install/SuperCollider/`
+    $> cd supercollider
+    $> mkdir build
+    $> cd build
+    $> cmake -G Xcode -DCMAKE_PREFIX_PATH=`brew --prefix qt@5.5` ..
+    $> cmake --build . --target install --config RelWithDebInfo
 
-You can see the available build options with ```cmake -LH```.
+Additional build options can be seen with `cmake -LH`.
 
-To install, you may move this to /Applications or use it in place from the build directory.
+If successful, this will install the application to the folder `build/Install/SuperCollider/`.  If you want to move it to /Applications, move the folder `SuperCollider` rather than just the application bundle.
 
 **Note**: You can also open the produced SuperCollider.xcodeproj in Xcode, and build the "Install" scheme in place of the last step. Do make sure you run the previous configuration steps.
 
+
 #### Step by step explanation of the Build instructions:
 
 ##### Create a `build` folder if one doesn't already exist:
 
-    mkdir -p build
-    cd build
+    $> mkdir build
+    $> cd build
+
+It is customary but not required to build SC in a subfolder `build` of the SC source. It has a few advantages:
+
+- `git` can be used to control the source while building
+- pointing to the source in the cmake configuration step just requires `..`
 
-##### Prepare for building by making a configuration file:
+*Note:* SC source subfolders starting with `build` are ignored by git.
```

Comment:
I think this (from "It is customary ..." to this line) is a little too detailed.